snackpub/src/index.ts

```diff
@@ -160,3 +136,89 @@ async function runAsync() {
     process.exit(1);
   }
 })();
+
+/**
+ * Register process signals to shutdown the server
+ */
+function registerShutdownHandlers(server: Server, redisClients: NullableRedisClientType[]) {
+  const shutdownSignals: NodeJS.Signals[] = [
+    'SIGHUP',
+    'SIGINT',
+    'SIGTERM',
+    'SIGUSR2', // Nodemon sends SIGUSR2 when it restarts the process it's monitoring
+  ];
+
+  for (const signal of shutdownSignals) {
+    process.once(signal, (signal: NodeJS.Signals) => {
+      console.log(
+        `Received ${signal}; the HTTP server is shutting down and draining existing connections`
+      );
+      shutdownAsync(server, redisClients);
+    });
+  }
+}
+
+/**
+ * Register Redis client handlers
+ */
+function registerRedisClientErrorHandlers(redisClients: NullableRedisClientType[]) {
+  for (const redisClient of redisClients) {
+    redisClient?.on('error', async (error) => {
+      // node-redis emits errors even when reconnecting,
+      // leave this error handler as no-op and handle in `reconnectStrategy`.
+      debug('Redis client error', error);
+    });
+    redisClient?.on('reconnecting', () => {
+      debug(`Reconnecting Redis server in ${REDIS_RECONNECT_RETRY_DELAY_MS}ms`);
```

Comment:
```suggestion
      debug(`Reconnecting to Redis server in ${REDIS_RECONNECT_RETRY_DELAY_MS}ms`);
```

Also, it might be helpful to include information about which Redis server we're reconnecting to, if that info is easily available.